Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/sdk/common/azure/serverless/AzureSparkServerlessCluster.java

```diff
@@ -517,6 +533,63 @@ public AzureHttpObservable getHttp() {
                 .defaultIfEmpty(this);
     }
 
+    @Nullable
+    private UpdateSparkResourcePool preparePatchResourcePool(int workerTargetInstanceCount) {
+        if (master == null || worker == null) {
+            return null;
+        }
+
+        UpdateSparkResourcePool patchBody = new UpdateSparkResourcePool();
+
+        return patchBody
+                .withName(getName())
+                .withProperties(new UpdateSparkResourcePoolParameters()
+                        .withSparkResourceCollection(Arrays.asList(
+                                new UpdateSparkResourcePoolItemParameters()
+                                        .withName(SparkNodeType.SPARK_WORKER)
+                                        .withTargetInstanceCount(workerTargetInstanceCount)
+                        )));
+    }
+
+    private Observable<SparkResourcePool> updateResourcePoolRequest(int workerTargetInstanceCount) {
+        URI uri = getUri();
+
+        UpdateSparkResourcePool patchBody = preparePatchResourcePool(workerTargetInstanceCount);
+        if (patchBody == null) {
+            return Observable.error(new IllegalArgumentException("Spark master and worker is not stable yet. Please retry until they are stable."));
```

Comment:
Should be a special Type exception.

Reply:
@wezhang@outlook.com Do you sugggest we just use `RuntimeException` or define a new `SparkServerlessClusterException` extends from `RuntimeException`?

---
In reply to: [197367947](https://github.com/Microsoft/azure-tools-for-java/pull/1734#discussion_r197367947) [](ancestors = 197367947)